### PR TITLE
#825: Allow passing explicit connection to '_PropertyMixin.{reload,patch}'.

### PR DESCRIPTION
**Pull request by tseaver**
See #825.

### CODE REVIEW

**Comment by tseaver** [comment]
@dhermes PTAL

File: gcloud/storage/_helpers.py [hunk 1]
```diff
@@ -52,12 +52,20 @@ def __init__(self, name=None):
...
+    def reload(self, connection=None):
+        """Reload properties from Cloud Storage.
+
+        :type connection: :class:`gcloud.storage.connection.Connection`
+        :param connection: An explicit connection to use for the API request.
+                           If not passed, use the connection assigned to
+                           the object in its constructor.
+        """
+        if connection is None:
+            connection = self.connection
```
Comment (dhermes) [inline gcloud/storage/_helpers.py]: See #843.

```python
from gcloud.storage._helpers import _require_connection
connection = _require_connection(connection)
```

Comment (tseaver) [inline gcloud/storage/_helpers.py]: I had thought we might make a subsequent pass, removing all the `self.connection` usage (and the existence of a connection attribute) throughout the three APIs.  I can do the change here, instead, if you think that best.

Comment (dhermes) [inline gcloud/storage/_helpers.py]: That just makes us change this line twice instead of once. I prefer `_require_connection` once (and for all).

**Comment by tseaver** [comment]
FWIW forcing use of `_require_connection` inside `_PropertyMixin.{reload,patch}` makes this a bigger PR:  we have to chase all the tests in derived classes which don't set up the implicit default.

**Comment by tseaver** [comment]
I was wrong there:  the test failures were due to `storage.api.get_bucket`, which wasn't passing through the explicit connection to `bucket.reload`.

**Comment by tseaver** [comment]
@dhermes Rebased w/ protobuf version pin fix, and changed to use `_require_connection`. PTAL

File: gcloud/storage/_helpers.py [hunk 2]
```diff
@@ -33,11 +33,6 @@ class _PropertyMixin(object):
     """
 
     @property
-    def connection(self):
```
Comment (dhermes) [inline gcloud/storage/_helpers.py]: Pleasantly surprised this can be axed. Can't be removed from `Blob/Bucket` until we get `ACL` stuff finished?

**Comment by dhermes** [comment]
LGTM

**Pull request merged**